src/analysis/Environments/DeleteAppModal.ts

```diff
@@ -21,6 +22,7 @@ export interface DeleteAppModalProps {
 
 export const DeleteAppModal = (props: DeleteAppModalProps): ReactNode => {
   const { app, onDismiss, onSuccess, deleteProvider } = props;
+  const { withErrorReportingInModal } = withErrorReporter(useNotificationsFromContext());
```

Comment:
Nit: I think it might read a bit more clearly/conventional if the hook is called in its own statement.

```suggestion
  const notifications = useNotificationsFromContext();
  const { withErrorReportingInModal } = withErrorReporter(notifications);
```

But given how frequently this pair of functions appears, it may also be worth encapsulating in a hook `useErrorReporting` or something like that.

Reply:
hmmm... I think we were going for a one-liner when we last looked at the code on my local.... and I am a bit nervous of hiding "....FromContext" in a secondary hook like useErrorReporting.... 

As I have it, it's a one-liner and the context usage is obvious... expressive is good.

Reply:
> I am a bit nervous of hiding "....FromContext" in a secondary hook like useErrorReporting....

Why? `...fromContext` hooks will fail loudly if they're misused (used outside of the required context provider). The wrapper could be called `useErrorReportingFromContext` or something like that if desired, and would reduce the repeated boilerplate.

But in any case, this was more a suggestion for future consideration vs a request for changes to this PR.